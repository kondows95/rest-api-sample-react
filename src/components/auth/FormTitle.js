import React from 'react'
import PropTypes from 'prop-types'
import Box from '@material-ui/core/Box'
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';

const FormTitle = React.memo(({children}) => {
  return (
    <Box display="flex" flexDirection="column" alignItems="center">
      <Box width={40} height={40} borderRadius="50%" bgcolor="secondary.main">
        <Box m={1} color="secondary.contrastText">
          <LockOutlinedIcon />
        </Box>
      </Box>
      <Box fontSize="h5.fontSize" mt={1} mb={2}>
        {children}
      </Box>
    </Box>
  )
})

FormTitle.propTypes = {
  children: PropTypes.node.isRequired
}
export default FormTitle;

