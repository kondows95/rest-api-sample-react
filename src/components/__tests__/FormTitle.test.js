import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import FormTitle from '../auth/FormTitle';

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

describe("FormTitle component snapshot", () => {
  it('matches the snapshot', () => {
    const FormTitleSnapshot = renderer.create(
      <FormTitle
        children={"Please enter your email"}
      />
    ).toJSON();
    expect(FormTitleSnapshot).toMatchSnapshot();
  })
})
