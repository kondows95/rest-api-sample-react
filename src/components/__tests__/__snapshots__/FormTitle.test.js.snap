// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTitle component snapshot matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0V0z"
          />
          <path
            d="M0 0h24v24H0V0z"
            opacity=".87"
          />
        </g>
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-14"
  >
    Please enter your email
  </div>
</div>
`;
